zebra-network/src/meta_addr.rs

```diff
@@ -656,32 +739,83 @@ impl Ord for MetaAddr {
         use std::net::IpAddr::{V4, V6};
         use Ordering::*;
 
-        let oldest_first = self.get_last_seen().cmp(&other.get_last_seen());
-        let newest_first = oldest_first.reverse();
-
-        let connection_state = self.last_connection_state.cmp(&other.last_connection_state);
-        let reconnection_time = match self.last_connection_state {
-            Responded => oldest_first,
-            NeverAttemptedGossiped => newest_first,
-            NeverAttemptedAlternate => newest_first,
-            Failed => oldest_first,
-            AttemptPending => oldest_first,
-        };
-        let ip_numeric = match (self.addr.ip(), other.addr.ip()) {
+        // First, try states that are more likely to work
+        let responded_never_failed_attempting =
```

Comment:
I'm not sure if it's just me, but the name seems a bit confusing. I know naming is hard, and to be honest I don't have any good suggestions. The closest I could come up with is something like `most_likely_state_to_succeed` :/

Reply:
I went with `more_reliable_state`, because that's almost exactly what it is:
- `Responded` peers are more reliable than (most) `NeverAttempted...` peers
- `NeverAttempted...` peers are more reliable than (most) `Failed` peers

See commit 96dc93bf

Reply:
Ok, that makes sense :+1: